packages/client/src/runtime/RequestHandler.ts

```diff
@@ -189,6 +188,11 @@ export class RequestHandler {
 
   handleRequestError({ error, clientMethod, callsite }: HandleErrorParams): never {
     debug(error)
+    // TODO: This is a workaround to keep backwards compatibility with clients
+    // consuming NotFoundError
+    if (error instanceof NotFoundError) {
```

Comment:
Will this be made a PrismaClientKnownRequestError in the future?

Reply:
Found the answer here https://github.com/prisma/prisma/pull/16162/files#diff-99659f0376b81639e19aee7c9cf5dee20ca6f2a4d68cdf549b2439940466d5f3R14

Reply:
Yes, that's the intention. For the moment the NotFoundError became deprecated, it inherited from `PrismaClientKnownRequestError` thus getting a code and the rest of its properties, and once we reach the next major version it should be replaced.